### PR TITLE
Create command to trigger compile / pre_release / release from main

### PR DESCRIPTION
**Pull request by carmelachivers1**

### CODE REVIEW

**Review by HughParsonage — DISMISSED** [review]
Good. Only major issue is for trailing whitespace. We may want to consider `truthy` values but very much optional. Other comment is for style.

File: tests/testthat/test_release_status.R [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+})
+
+#only false when compile is present but is not returned. Not false if compile not present but is returned.
+
+test_that("returns pre_release when present", {
+  expect_equal(any(grepl("pre_release", release_status("release-status-compile.tex"))), TRUE)
+})
+
+test_that("returns release when present", {
+  expect_equal(any(grepl("^[release]", release_status("release-status-compile.tex"))), TRUE)
```
[the remark this answers is not in the document]
Comment (HughParsonage) [reply tests/testthat/test_release_status.R]: Yeah, regular expression are highly contextual.

File: R/release_status.R [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+
+release_status <- function(filename) {
+  lines <- read_lines(filename)
+  
+  lines_before_begin_document <-
+    lines[1:grep("\\begin{document}", lines, fixed = TRUE)]
+
+  if(any(grepl("^[%]\\s*[Rr]elease(:| [=])\\s*[Tt]rue$", lines_before_begin_document))){
```
[the remark this answers is not in the document]
Comment (carmelachivers1) [reply R/release_status.R]: Hm, good point. Does your expression require a full stop? That might be annoying for users. What about /s*$? (which I think means "some white space or no white space at the end of the line").

[the remark this answers is not in the document]
Comment (HughParsonage) [reply R/release_status.R]: No no, it's rendered wrong. Imagine typing `compile: true <space>` where `<space>` is a space. This wouldn't have any visible differences from `compile: true`, but your test would give different results.

[the remark this answers is not in the document]
Comment (carmelachivers1) [reply R/release_status.R]: Ah I see. \s*$ works, but maybe it's too restrictive anyway to say the command has to be on its own line...

**Comment by carmelachivers1** [comment]
What's codecov about?

File: R/release_status.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (HughParsonage) [reply R/release_status.R]: Ooh no I prefer `\s*$`. Nice!

[the remark this answers is not in the document]
Comment (HughParsonage) [reply R/release_status.R]: Alternatively you can use `trimws` when you define `lines_before_begin_document` to "**trim** **w**hite-**s**pace"

**Comment by HughParsonage** [comment]
> What's codecov about?

This determines how much of the package has been tested: each test in the `tests/testthat` suite is run and `codecov` works out which lines of the code were used in each test. The goal is to get every line tested by at least one test: "100% coverage".

At each pull request, `codecov` runs to determine whether the code to be merged will increase or decrease the percent of lines covered by tests. In this case, you've added extra code (more lines that require tests), but all of them are tested, so the coverage % increases.

**Comment by HughParsonage** [comment]
Questions: 
1. What should the function return when none of these directives are present?
2. What does the function currently return when none of these directives are present?

**Review by HughParsonage — APPROVED** [review]

**Pull request merged**